projects/admin/src/app/record/search-view/document-advanced-search-form/advanced-search-config.service.ts

```diff
@@ -0,0 +1,118 @@
+ * Copyright (C) 2021-2023 UCLouvain
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { Injectable } from '@angular/core';
+import { DocumentApiService } from '@app/admin/api/document-api.service';
+import { marker as _ } from '@biesbjerg/ngx-translate-extract-marker';
+import { Observable, of } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+import { IAdvancedSearchConfig, IFieldsData, ILabelValue, ILabelValueField, ISelectOptions } from './i-advanced-search-config-interface';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AdvancedSearchConfigService {
+
+  /** Configuration loaded from backend */
+  private configLoaded: boolean = false;
+
+  /** Fields options */
+  private fieldOptions: ILabelValue[] = [];
+
+  /** Fields data */
+  private fieldData: IFieldsData;
+
+  /** Fields mapping */
+  private fieldMappingMap: Map<string, string>;
+
+  /**
+   * Constructor
+   * @param documentApiService - DocumentApiService
+   */
+  constructor(private documentApiService: DocumentApiService) { }
+
+  /**
+   * Load advanced search configuration
+   * @returns an observable with a Boolean that gives
+   * information about the loaded configuration.
+   */
+  load(): Observable<boolean> {
+    if (this.configLoaded) {
+      return of(this.configLoaded);
+    }
+
+    return this.documentApiService.getAdvancedSearchConfig().pipe(
+      tap((config: IAdvancedSearchConfig) => this.process(config)),
+      tap(() => this.configLoaded = true),
+      map(() => true)
+    );
+  }
+
+  /**
+   * Get operator
+   * @returns operators used for forms
+   */
+  getOperators(): ISelectOptions[] {
+    return [
+        { label: _('and'), value: 'AND' },
+        { label: _('or'), value: 'OR' },
+        { label: _('and not'), value: 'AND NOT' }
+      ]
+  }
+
+  /**
+   * Field mapping
+   * @param field - the field name
+   * @returns the field config
+   */
+  fieldMapping(field: string): string {
+    if (!this.fieldMappingMap.has(field)) {
+      throw new SyntaxError(`Field mapping does not exist (${field})`);
+    }
+    return this.fieldMappingMap.get(field);
+  }
+
+  /**
+   * Get Fields config
+   * @returns - fields options
+   */
+  getFieldsConfig(): ILabelValue[] {
+    return this.fieldOptions;
+  }
+
+  /**
+   * Get fields data
+   * @returns - fields data options
+   */
+  getFieldsData(): IFieldsData {
+    return this.fieldData;
+  }
```

Comment:
isn't a little bit overkill to create a function to return a composant attributes ? just use "public" when you declare corresponding attributes

Reply:
I did this to avoid data modifications later on.